src/cascadia/TerminalCore/Terminal.hpp

```diff
@@ -291,6 +291,13 @@ class Microsoft::Terminal::Core::Terminal final :
         End = 0x2
     };
 
+    struct TextCopyData
+    {
+        std::wstring plainText;
+        std::optional<std::string> html;
+        std::optional<std::string> rtf;

```

Comment:
i wonder. it should be a fair assumption that if the string is empty, it didn't exist. do we need the optional?

(Not asking you to change this, just curious)

Reply:
That depends on user expectations. If I select the empty space after the prompt, I'd expect it to copy an empty string (which got trimmed) and put that into the clipboard. Also, the UI behaves like something is copied. (User has an active selection -> triggers copy action -> selection goes away -> ctrl+v -> should paste empty string). Now, if we couldn't _select_ the empty spaces then this would be a different story 😅

Reply:
That case is probably fine even if "empty string" means "not present"! Because the framing for the HTML and RTF documents **always** includes data (the DOCTYPE/Clipboard header or the RTF preamble), selecting and copying the empty space after the prompt wouldn't result in an empty string.

Reply:
> the HTML and RTF documents always includes data (the DOCTYPE/Clipboard header or the RTF preamble)

Sorry, I missed that part completely. That means we can use an empty string as an empty RTF/HTML data. 😄

Reply:
Except, we do return an empty string from the three text generation functions, but only when get a CopyRequest with `req.beg > req.end` i.e. empty selection range (possibly unreal). Since it's an empty selection, it should be okay to return an empty string I guess.

Should I replace it with assert calls to convey that the req is ill-formed? WDYT?

Reply:
Another option is to remove the special case handling and let them fill the std::string with just HTML/RTF  header. (Have to test whether this works fine or not)

Reply:
I am A-OK with us leaving the optionals, btw! It does make it a lot clearer.

Reply:
Sure, let's go ahead with `std::optional` approach 👍